Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,333 @@
+# Retrieving Relationship Information
+
+This proposal describes how relationship information can be generated from the DSL, introduces some concepts
+to give hints for relationship retrieval and adds mechanisms for manually defined hints.
+
+A potential REST interface exporter will use such relationship information to generate Hypermedia information.
+Specifically it will generate hyperlinks which SHOULD be place in a designated `_links` section.
+A REST exporter MAY also inline those links if it can ensure that no name collisions occur.  
+This document proposes that [RFC 6570 URI Template](https://tools.ietf.org/html/rfc6570) SHOULD be used to encode those links.
+
+The following chapters explain how such relationship information is generated.
+
+## 1. Using the path "structure"
+
+The path structure of the Webservice can be used to infer relationship information.  
+For example the fact that the `/user/:userid/post` path is a subroute to `/user/:userid` path can be used to 
+generate relationship information for every user, namely a relationship pointing to `user/{userid}/post`.
+
+In general every endpoint will relate to **all** of its children/subroutes.
+
+Giving an example: When a client sends a request for `/user/532` the a potential REST exporter will return the following:
+```json
+{
+  "id": "532",
+  "name": "Rudi",
+  "lastname": "Raser",
+  "email": "rudi@raser.de",
+  "_links": {
+    "self": "https://example.api/user/532",
+    "post": "https://example.api/post"
```

Comment:
Is this intentionally not `user/532/post`?

Reply:
You are right. That should be `"post": "https://example.org/user/532/post"`. Thanks for catching that 👍